### DISCUSSION TITLE
Some replacements not being replaced

### DISCUSSION THREAD

**Discussion by nullxx** [sequelize/sequelize#15410] [Help & Questions]
If you try to run a query like this one.

```sql
CREATE USER ?@? IDENTIFIED BY ?
```

Replacements
```js
['test', '%', '12345678']
```

The following function (`canPrecedeNewToken`) that gets called inside `injectReplacements`.
https://github.com/sequelize/sequelize/blob/1a47feaab37c261dce55741fa703ff86a39c365e/src/utils/sql.ts#L224-L226

https://github.com/sequelize/sequelize/blob/1a47feaab37c261dce55741fa703ff86a39c365e/src/utils/sql.ts#L266-L268

Because of `/[\s(>,=]/` the second replacement `@?` is not replaced because `previousChar` is '@'.

Is this an expected behavior? 
How could I do to handle this type of query?

**Comment by ephys** [comment]
It is the expected behavior but it is unfortunate. `?@?` looks like something that could be an operator so it's not getting replaced

Assuming you're allowed to put spaces around `@`, this would work:

```sql
CREATE USER ? @ ? IDENTIFIED BY ?
```

**Reply by nullxx** [reply]
What operator could be `?@?`?

```
CREATE USER 'test' @ '%' IDENTIFIED BY 'passsword';
```
throws
```
You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near ''%' IDENTIFIED BY 'passsword'' at line 1 
```

**Reply by ephys** [reply]
> What operator could be `?@?`?

There isn't such an operator AFAIK but `?` and `@` are both used in some operators. There is also the fact that our replacements injection engine must be really fast and must work in many different dialects, so it's only aware of the bare minimum (almost just strings and comments)

You can open a bug report for this issue. I also think [this feature request](https://github.com/sequelize/sequelize/issues/14299) should resolve your problem

In the meantime you can use this workaround:

```ts
const queryGenerator = sequelize.queryInterface.queryGenerator;

await sequelize.query(`CREATE USER ${queryGenerator.escape('test')}@${queryGenerator.escape('%')} IDENTIFIED BY ?`);
```

**Reply by nullxx** [reply]
Why do we not use '#' as a placeholder? Or another character that is not permitted in MySQL

**Reply by ephys** [reply]
There is an issue about that here: https://github.com/sequelize/sequelize/issues/15082
